Add server example (#9918)
huggingface/diffusers: examples/server/README.md

> ## OpenAI Compatible `/v1/images/generations` Server

Reviewer: ```suggestion
# Create a server
```

> This is a concurrent, multithreaded solution for running a server that can generate images using the `diffusers` library. This examples uses the Stable Diffusion 3 pipeline, but you can use any pipeline that you would like by swapping out the model and pipeline to be the ones that you want to use.

Reviewer: ```suggestion
Diffusers' pipelines can be used as an inference engine for a server. It supports concurrent and multithreaded requests to generate images that may be requested by multiple users at the same time.

This guide will show you how to use the [`StableDiffusion3Pipeline`] in a server, but feel free to use any pipeline you want.
```

> ### Installing Dependencies

Reviewer: ```suggestion
```

> Start by going to the base of the repo and installing it with:
> ``py
> pip install .
> ```

Reviewer: ```suggestion
Start by navigating to the `examples/server` folder and installing all the dependencies.

``py
pip install .
pip install -f requirements.txt
```
```

> The pipeline can then have its dependencies installed with:
> ```py
> pip install -f requirements.txt
> ```

Reviewer: ```suggestion
```

> ### Running the server

Reviewer: Would be nice to briefly go over the `server.py` script and highlight the important bits. For example, how is multithreading enabled? Adding more details like this helps users better understand whats happening under the hood.

```suggestion
```

> This server can be run with:
> ```py
> python server.py
> ```
> The server will be spun up at http://localhost:8000. You can `curl` this model with the following command:
> ```
> curl -X POST -H "Content-Type: application/json" --data '{"model": "something", "prompt": "a kitten in front of a fireplace"}' http://localhost:8000/v1/images/generations
> ```

Reviewer: ```suggestion
Launch the server with the following command.

```py
python server.py
```

The server is accessed at http://localhost:8000. You can `curl` this model with the following command.

```
curl -X POST -H "Content-Type: application/json" --data '{"model": "something", "prompt": "a kitten in front of a fireplace"}' http://localhost:8000/v1/images/generations
```

```

> ### Upgrading Dependencies

Reviewer 2: Can we also add some notes on the multi-threaded nature of this example and show some GPU utilization numbers with and without threading?

Author: Yes. I'll add that in the next commit.

> ### Upgrading Dependencies

Reviewer: ```suggestion
```

> If you need to upgrade some dependencies, you can do that with either [pip-tools](https://github.com/jazzband/pip-tools) or [uv](https://github.com/astral-sh/uv). With `uv`, this looks like:
> ```
> uv pip compile requirements.in -o requirements.txt

Reviewer: ```suggestion
If you need to upgrade some dependencies, you can use either [pip-tools](https://github.com/jazzband/pip-tools) or [uv](https://github.com/astral-sh/uv). For example, upgrade the dependencies with `uv` using the following command.

```
uv pip compile requirements.in -o requirements.txt
```
